src/data-sources/hooks/useDataSources.ts

```diff
@@ -17,16 +17,13 @@ export const useDataSources = < SourceConfig extends DataSourceConfig = DataSour
 		useDispatch< NoticeStoreActions >( noticesStore );
 	const { goToMainScreen } = useSettingsContext();
 
-	const checkDisplayNameConflict = useCallback(
-		( displayName: string, uuid: string ): boolean => {
-			if ( ! displayName ) {
-				return false;
-			}
-			const existingSource = dataSources.find(
+	const canUseDisplayName = useCallback(
+		( displayName: string, uuid: string ): boolean =>
+			// ensure display name isn't an empty string or already in use by another data source
+			! displayName ||
```

Comment:
I think this logic is wrong. An empty string returns `true`

Reply:
```
const canUseDisplayName = ( displayName: string, uuid: string ) =>
	displayName && dataSources.every(
		source => source.uuid === uuid || source.service_config.display_name !== displayName
	);
```

Reply:
You're right, thank you! Updated here: [`38b4dec` (#295)](https://github.com/Automattic/remote-data-blocks/pull/295/commits/38b4dec618ca09f64c8760cc06ecf80db98d84e5)